### ISSUE TITLE
Unable to load user control from C++/WinRT project into a WinUI3 desktop app using .NET5 projection

### ISSUE THREAD

**Issue by Rochishaa** [microsoft/CsWinRT#893]
**Describe the bug**
I have created a C++/WinRT component, _NewComponent_, that contains a simple user control called _MyControl_. I then created a .NET5 projection out of it - _NewProj_ . I now want to load this user control in a WinUI3 desktop app - _NewApp_. 

Here is my MainWindow.xaml : 

`<Window
    x:Class="NewApp.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:NewApp"
    xmlns:controls="using:NewComponent"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d">
    <StackPanel Orientation="Horizontal" HorizontalAlignment="Center" VerticalAlignment="Center">
        <Button x:Name="myButton" Click="myButton_Click">Click Me</Button>
        <controls:MyControl/>
    </StackPanel>
</Window> `

**To Reproduce**
1. Build _NewComponent.sln_ to generate nuget.
2. Add generated nuget to _NewApp_ project. Run the application.

**Expected behavior**
_NewApp_ should show **MyControl** from _NewComponent_ . But I don't see **MyControl** being loaded.

**Source repo**
https://github.com/Rochishaa/UserControlWinUI3

**Comment by asklar** [comment]
@alwu-msft in case this sounds familiar

**Comment by Rochishaa** [comment]
I am now getting the following runtime exception when launching my WinUI3 desktop app - NewApp :

> Microsoft.UI.Xaml.Markup.XamlParseException
  HResult=0x802B000A
  Message=XAML parsing failed.
  Source=WinRT.Runtime
  StackTrace:
   at WinRT.ExceptionHelpers.ThrowExceptionForHR(Int32 hr)
   at ABI.Microsoft.UI.Xaml.IApplicationStatics.global::Microsoft.UI.Xaml.IApplicationStatics.LoadComponent(Object component, Uri resourceLocator, ComponentResourceLocation componentResourceLocation)
   at Microsoft.UI.Xaml.Application.LoadComponent(Object component, Uri resourceLocator, ComponentResourceLocation componentResourceLocation)
   at NewApp.MainWindow.InitializeComponent() in C:\Dev\UserControlWinUI3\WinUI3MainApp\NewApp\NewApp\obj\x64\Debug\net5.0-windows10.0.19041.0\MainWindow.g.i.cs:line 39
   at NewApp.MainWindow..ctor() in C:\Dev\UserControlWinUI3\WinUI3MainApp\NewApp\NewApp\MainWindow.xaml.cs:line 28

**Comment by angelazhangmsft** [comment]
I was able to repro this and was not seeing any issues with a C++/WinRT app referencing a C++/WinRT component with a WinUI user control. So the issue seems to be specific to C#/.NET 5 applications which reference a C++/WinRT component with WinUI user control. I was also able to repro the issue with a project reference in addition to package reference.

We will need to do some further investigation. 

[callstack_893.txt](https://github.com/microsoft/CsWinRT/files/6819394/callstack_893.txt)

**Comment by LadderLogic** [comment]
@angelazhangmsft any projected release date or known workaround for this? We are trying to a project WinRT C++ Win UI 3 control in a Win UI 3 C# app.

**Comment by angelazhangmsft** [comment]
@LadderLogic - we looked into this issue, and it is a known issue for WinAppSDK 1.0. You can workaround it by adding the following target to the end (after all imports of .targets files) of your Windows Runtime Component's .vcxproj:

```
  <Target Name="GetPriIndexName">
    <PropertyGroup>
       <!-- Winmd library targets use the default root namespace of the project for the App package name -->
       <PriIndexName Condition="'$(RootNamespace)' != ''">$(RootNamespace)</PriIndexName>
       <!-- If RootNamespace is empty fall back to TargetName -->
       <PriIndexName Condition="$(PriIndexName) == ''">$(TargetName)</PriIndexName>
     </PropertyGroup>
  </Target>
```

There is currently not an ETA for a fix, we are considering for 1.0 servicing but I suggest using the workaround for now. Closing this as a duplicate of https://github.com/microsoft/WindowsAppSDK/issues/1769 - you can follow that issue for updates.

**Comment by LadderLogic** [comment]
@angelazhangmsft Thanks for the update and workaround. We used some inspiration from 
[Windows Terminal](https://github.com/Corillian/WindowsTerminal/tree/main/Microsoft.Terminal.Control.Projection) project to move forward with our projection, but your suggested workaround looks cleaner. We will adapt until the official fix. 🙏
